### DISCUSSION TITLE
Yaapu Lua script

### DISCUSSION THREAD

**Discussion by ardisd** [EdgeTX/edgetx#2078] [Q&A] [answered]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

There appears to be a bug running the yaapu Lua script or edgeTX 2.7.1. The widget keeps restarting on my RadioMaster 16TX MKII

### Expected Behavior

The Lua script should display telemetry data and not restart

### Steps To Reproduce

The issue happens when the script executes

### Version

2.7.1

### Transmitter

Radiomaster TX16S

**Comment by athertop** [comment]
The yaapu widget config screen uses gVar 9 of FM8 to record a flag telling the widget if some config changed. If the widget sees gVar 9 of FM8 != 0 then it does the reinitialise you are seeing then tries to set the value back to 0 - but it fails to do any of this because of a recent commit in EdgeTx which affects how lua updates gVars. This is the commit that fixed a longstanding issue in OpenTx, but caused the issue for yaapu: https://github.com/EdgeTX/edgetx/commit/0391174e066a8b28cb87c656bec787a4a0fb648c
I have spoken with Alex (yaapu dev) and he is aware of this issue. I am sure he will address the issue inside the widget code. As this is not specifically an EdgeTx issue, it should not really be reported here.
I have also written a post all about this in RCG here: https://www.rcgroups.com/forums/showpost.php?p=49299729&postcount=852
This includes how to work around the issue - its a very simple config change you will need to do for each model that uses the yaapu widget.

**Comment by AndreHenke** [comment]
Yaapu telemetry is not working on Edge TX.
I reinstalled yaapu on my TX16s, and screen goes black.
I tried to change the theme, but half of the information does not appear.
Should I go back to Open TX??
Thanks you!

**Comment by rotorman** [comment]
Just tested, works fine. Setup:
* RadioMaster TX16S mkII with EdgeTX v2.7.1 release.
* Yaapu LUA telemetry widget (**dev-branch !!!**)
* ArduPilot with MAVLink telemetry
* Mav2PT converter

Video proof: https://www.youtube.com/watch?v=CTjDNwXNxrY

**Comment by rotorman** [comment]
From YT video https://www.youtube.com/watch?v=CTjDNwXNxrY comment I take it, the issue for @AndreHenke is solved:
![grafik](https://user-images.githubusercontent.com/21011587/174738425-2e6e4a2a-5a48-4077-b790-0f4f7c50121e.png)

@ardisd Can the issue be closed?

**Comment by athertop** [comment]
> Yaapu telemetry is not working on Edge TX. I reinstalled yaapu on my TX16s, and screen goes black. I tried to change the theme, but half of the information does not appear. Should I go back to Open TX?? Thanks you!

What RC system are you using? FfSky? Crsf? Dragon link?

**Comment by AndreHenke** [comment]
Hi!
It's TX16S and the receiver is Frsky. It was working fine on OpenTx. I changed to EdgeTX and it stopped working. I downloaded the new version for horus and it went black. I downloaded version 1.95 and it appears to be working, but it is not as good as the older version. It doesn't show satellites as it did before. Also layout is a little confusing. Can you give me the correct link for Yaapu files for Radiomaster TX16S for EdgeTx?
Thanks so much!
Rgds,
Andre

**Comment by athertop** [comment]
For EdgeTX you'll need to download the current dev version 1.9.6. Select the dev branch in Github before downloading

**Reply by GhanshyamNaik** [reply]
Please can you provide the link where i can find the dev version 1.9.6 of the Yaapu Script. I am not able to find it.

**Reply by rotorman** [reply]
Nowadays you just need the master branch (no need to resort to old 1.9.6). Get latest version from: https://github.com/yaapu/FrskyTelemetryScript

**Reply by GhanshyamNaik** [reply]
Actually I didnt understand your answer. To clarify i have downloaded 2 different zip files. One is the "FrskyTelemetryScript-Horus-Widget-1.9.5" which is the latest version in the "Releases" section. Second is the "FrskyTelemetryScript-master". If you are referring to the second master folder to be the one required for EdgeTx then the problem is that i am not able to find the SD card contents in the master zip file.
Sorry but i am relatively new to all these Radio and EdgeTx stuff.

I have Radiomaster TX16S with EdgeTX 2.7.1.
It would be great if you can guide me with which files would be needed from the master zip (if that is what you are suggesting in you original comment) file for my configuration

**Reply by rotorman** [reply]
See: https://github.com/yaapu/FrskyTelemetryScript/wiki/Installation-on-Horus-radios

**Comment by rotorman** [comment]
@AndreHenke I understood from your YouTube comment that you got it working already. 1.9.5 is not correct.
To git clone the dev branch: `git clone -b dev https://github.com/yaapu/FrskyTelemetryScript.git YaapuLUA`

**Comment by AndreHenke** [comment]
I installed 1.96 as directed on the page and it is working, but not as good as it was on the previous version. Maybe there are some bugs that are being handled now and will be corrected soon. Will keep watching!
Thank you both!

**Comment by jjsueldo** [comment]
I have the same problem with my Jumper T16
installed EDGETX 2.7.0.  and Yaapu 1.9.5 (last version on official page)

the screen shows a few characters and graphics and freezes

have already found a solution to this problem ???

**Reply by pfeerick** [reply]
1.9.5 is not the correct version... 

From the [official page](https://github.com/yaapu/FrskyTelemetryScript) - you want to click on the link where it says to download 1.9.6

![image](https://user-images.githubusercontent.com/5500713/196830677-7791f111-055c-4770-8771-6999a3fd48fd.png)

**Comment by AndreHenke** [comment]
It's 1.96!

**Reply by pfeerick** [reply]
In your case (I was replying to someone else, who is running 1.9.5 ;)), it might be worth trying to download the updated version of the dev branch. yaapu is talking about there being a test 2.0.0 version in another issue, which according to the feedback is working ok => https://github.com/yaapu/FrskyTelemetryScript/issues/172

**Comment by AndreHenke** [comment]
You are correct!

**Comment by jjsueldo** [comment]
Thank ´s ALL for the clarification, yesterday a friend gave me his SD to copy the new version of Yaapu
I couldn't find it because 1.9.6 version is not in the list of releases (maybe they should include it in that list)

Regards

**Reply by athertop** [reply]
I believe 1.9.6 is in the Yaapu development branch only at this point, so you would need to select that in the branch drop-down before downloading the repo zip file

**Comment by pfeerick** [comment, marked as answer]
At the time this problem was posted, the solution was to use v1.9.6 (or a development version) of the Yaapu Telemetry Script and Widget. In the two years since, v2.0.0 (or later) appears to be the recommended version to use. At the time of writing, June 2024, the install process is to go to the [Yappu Telemetry Script and Widget github page](https://github.com/yaapu/FrskyTelemetryScript/), and download the EdgeTX compatible version from the from there (don't go into the Releases section, the link you want is there on the front page). Then, for colour screen radios, follow [these instructions](https://github.com/yaapu/FrskyTelemetryScript/wiki/Installation-on-Horus-radios). If you are using a B&W radio and have come across these instructions, you want [these instructions](https://github.com/yaapu/FrskyTelemetryScript/wiki/Installation-on-Taranis-radios) instead. 

Credit to those who commented in this thread, particularly [rotorman](https://github.com/rotorman), for the content going into this answer.
